### PR TITLE
Add more JUnit tests to Parser class for public parse methods

### PR DESCRIPTION
**Pull request by warrencxw**

### CODE REVIEW

File: src/test/java/seedu/splitlah/parser/ParserTest.java [hunk 1]
```diff
@@ -829,4 +829,197 @@ void parseCostList_delimiterExistsArgumentValid_doubleArrayContainingCostList()
...
+                "/sid 3 /n Lunch /p Alice /i Alice Bob Charlie /co 15 /gst -1 /sc 10";
+        try {
+            int output = Parser.parseGst(argumentWithIntegerArgumentUnderRange);
+            fail();
+        } catch (InvalidFormatException exception) {
+            String errorMessage = Message.ERROR_PARSER_INVALID_GST_SURCHARGE + Parser.GST_DELIMITER;
+            assertEquals(errorMessage, exception.getMessage());
+        }
+
+        // Test negative values
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/parser/ParserTest.java:901]: Possible for this to be "Test values larger than 100"

File: src/test/java/seedu/splitlah/parser/ParserTest.java [hunk 2]
```diff
@@ -829,4 +829,197 @@ void parseCostList_delimiterExistsArgumentValid_doubleArrayContainingCostList()
...
+                "/sid 3 /n Lunch /p Alice /i Alice Bob Charlie /co 15 /gst 7 /sc -1";
+        try {
+            int output = Parser.parseServiceCharge(argumentWithIntegerArgumentUnderRange);
+            fail();
+        } catch (InvalidFormatException exception) {
+            String errorMessage = Message.ERROR_PARSER_INVALID_SERVICE_CHARGE + Parser.SERVICE_CHARGE_DELIMITER;
+            assertEquals(errorMessage, exception.getMessage());
+        }
+
+        // Test negative values
```
Comment (ivanaitzliddat) [inline src/test/java/seedu/splitlah/parser/ParserTest.java:997]: Same comment as above

**Review by ivanaitzliddat — APPROVED** [review]
Looks good overall, but left comments

**Review by froststein — APPROVED** [review]
Looks good to merge.

**Pull request merged**